gcp/observability/logging_test.go

```diff
@@ -0,0 +1,877 @@
+		fle.mu.Unlock()
+		t.Fatalf("error in logging entry list comparison %v", err)
+	}
+
+	fle.entries = nil
+	fle.mu.Unlock()
+
+	// Make a streaming RPC. This should cause Log calls on the MethodLogger.
+	stream, err := ss.Client.FullDuplexCall(ctx)
+	if err != nil {
+		t.Fatalf("ss.Client.FullDuplexCall failed: %f", err)
+	}
+
+	stream.CloseSend()
+	if _, err = stream.Recv(); err != io.EOF {
+		t.Fatalf("unexpected error: %v, expected an EOF error", err)
+	}
+	grpcLogEntriesWant = make([]*grpcLogEntry, 4)
+	grpcLogEntriesWant[0] = &grpcLogEntry{
+		Type:        eventTypeClientHeader,
+		Logger:      loggerClient,
+		ServiceName: "grpc.testing.TestService",
+		MethodName:  "FullDuplexCall",
+		Authority:   ss.Address,
+		SequenceID:  1,
+		Payload: payload{
+			Metadata: map[string]string{},
+		},
+	}
+	grpcLogEntriesWant[1] = &grpcLogEntry{
+		Type:        eventTypeClientHalfClose,
+		Logger:      loggerClient,
+		ServiceName: "grpc.testing.TestService",
+		MethodName:  "FullDuplexCall",
+		SequenceID:  2,
+		Authority:   ss.Address,
+	}
+	grpcLogEntriesWant[2] = &grpcLogEntry{
+		Type:        eventTypeServerHeader,
+		Logger:      loggerClient,
+		ServiceName: "grpc.testing.TestService",
+		MethodName:  "FullDuplexCall",
+		SequenceID:  3,
+		Authority:   ss.Address,
+		Payload: payload{
+			Metadata: map[string]string{},
+		},
+	}
+	grpcLogEntriesWant[3] = &grpcLogEntry{
+		Type:        eventTypeServerTrailer,
+		Logger:      loggerClient,
+		ServiceName: "grpc.testing.TestService",
+		MethodName:  "FullDuplexCall",
+		Authority:   ss.Address,
+		SequenceID:  4,
+		Payload: payload{
+			Metadata: map[string]string{},
+		},
+	}
+	fle.mu.Lock()
+	if err := cmpLoggingEntryList(fle.entries, grpcLogEntriesWant); err != nil {
+		fle.mu.Unlock()
+		t.Fatalf("error in logging entry list comparison %v", err)
+	}
+	fle.mu.Unlock()
+}
+
+func (s) TestServerRPCEventsLogAll(t *testing.T) {
+	fle := &fakeLoggingExporter{
+		t: t,
+	}
+	defer func(ne func(ctx context.Context, config *config) (loggingExporter, error)) {
+		newLoggingExporter = ne
+	}(newLoggingExporter)
+
+	newLoggingExporter = func(ctx context.Context, config *config) (loggingExporter, error) {
+		return fle, nil
+	}
+
+	serverRPCEventLogAllConfig := &config{
+		ProjectID: "fake",
+		CloudLogging: &cloudLogging{
+			ServerRPCEvents: []serverRPCEvents{
+				{
+					Methods:          []string{"*"},
+					MaxMetadataBytes: 30,
+					MaxMessageBytes:  30,
+				},
+			},
+		},
+	}
+	cleanup, err := setupObservabilitySystemWithConfig(serverRPCEventLogAllConfig)
+	if err != nil {
+		t.Fatalf("error setting up observability %v", err)
+	}
+	defer cleanup()
+
+	ss := &stubserver.StubServer{
+		UnaryCallF: func(ctx context.Context, in *grpc_testing.SimpleRequest) (*grpc_testing.SimpleResponse, error) {
+			return &grpc_testing.SimpleResponse{}, nil
+		},
+		FullDuplexCallF: func(stream grpc_testing.TestService_FullDuplexCallServer) error {
+			for {
+				_, err := stream.Recv()
+				if err == io.EOF {
```

Comment:
As above; send a message?

Also it seems worthwhile to send a message from the client in both cases.

Reply:
Also, something I just realized is this takes away having to change this test in the future due to that bin logging bug of logging Server Headers on Trailers only response. Luckily we have it clearly documented in other PRs test and this PRs history.